miniwin/miniwin/src/miniwin_d3drmviewport.cpp

```diff
@@ -86,13 +76,22 @@ void Direct3DRMViewportImpl::PushVertices(const PositionColorVertex* vertices, s
 	SDL_ReleaseGPUTransferBuffer(m_device, transferBuffer);
 }
 
-HRESULT Direct3DRMViewportImpl::Render(IDirect3DRMFrame* group)
+HRESULT Direct3DRMViewportImpl::Render(IDirect3DRMFrame* frame)
 {
 	if (!m_device) {
 		return DDERR_GENERIC;
 	}
+	Direct3DRMFrameImpl* frameImpl = static_cast<Direct3DRMFrameImpl*>(frame);
+	if (!frameImpl) {
+		return DDERR_GENERIC;
+	}
 
-	CollectSceneData(group);
+	m_backgroundColor = frameImpl->m_backgroundColor;
+
+	std::vector<PositionColorVertex> vertices =
+		{{-1, -1, 0, 0, 255, 0, 255}, {1, -1, 0, 0, 0, 255, 255}, {0, 1, 0, 255, 0, 0, 128}};
+
+	PushVertices(vertices.data(), vertices.size());
```

Comment:
I wanted to keep this in a separate method to avoid a big hard to follow function ad more parts of the render process gets implemented.

Reply:
I'll revert this.
Are we going to upload all geometry to the gpu on every render?

Reply:
Hopefully we can figure out how to avoid that, but for now I don't know how else to do it.

Reply:
Makes sense :)